app/src/androidTest/java/com/github/h3lp3rs/h3lp/HelpParametersActivityTest.kt

```diff
@@ -1,106 +1,218 @@
 package com.github.h3lp3rs.h3lp
 
+import android.Manifest
 import android.app.Activity
 import android.app.Instrumentation
+import android.content.Context
 import android.content.Intent
+import android.location.Location
 import android.net.Uri
-import androidx.test.espresso.action.ViewActions
-import androidx.test.espresso.assertion.ViewAssertions
-import androidx.test.espresso.intent.Intents
-import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents.*
+import androidx.test.espresso.intent.matcher.IntentMatchers.*
+import androidx.test.espresso.matcher.ViewMatchers.*
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.GrantPermissionRule
+import com.github.h3lp3rs.h3lp.LocalEmergencyCaller.DEFAULT_EMERGENCY_NUMBER
+import com.github.h3lp3rs.h3lp.locationmanager.GeneralLocationManager
+import com.github.h3lp3rs.h3lp.locationmanager.LocationManagerInterface
+import com.github.h3lp3rs.h3lp.preferences.Preferences
+import com.github.h3lp3rs.h3lp.preferences.Preferences.Companion.Files.PRESENTATION
+import com.github.h3lp3rs.h3lp.preferences.Preferences.Companion.USER_AGREE
+import com.github.h3lp3rs.h3lp.preferences.Preferences.Companion.clearAllPreferences
 import org.hamcrest.Matchers.*
-
+import org.junit.After
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.anyOrNull
+import org.mockito.Mockito.`when` as When
 
-import androidx.test.espresso.Espresso.onView
-import androidx.test.espresso.matcher.ViewMatchers.*
 
+// Case example of a possible query when a user clicks on the call for emergency button
+private val CORRECT_EMERGENCY_CALL = Triple(6.632, 46.519, "144")
 
 @RunWith(AndroidJUnit4::class)
 class HelpParametersActivityTest {
+    private val locationManagerMock: LocationManagerInterface =
+        mock(LocationManagerInterface::class.java)
+    private val locationMock: Location = mock(Location::class.java)
 
     @get:Rule
     val testRule = ActivityScenarioRule(
         HelpParametersActivity::class.java
     )
 
+
+    @get:Rule
+    var mRuntimePermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION)
+
+    @Before
+    fun setUp() {
+        init()
+        clearAllPreferences(ApplicationProvider.getApplicationContext())
+        Preferences(PRESENTATION, ApplicationProvider.getApplicationContext()).setBool(
```

Comment:
You can change this and use our wonderful new local storage class (depending on who merges first)

Reply:
Setting the static context variable will be needed as well (see Sign in/Irrelevant presentation tests)